app/src/main/java/com/example/khalk/network/TestCursorAdapter.java

```diff
@@ -0,0 +1,302 @@
+    private ProgressBar loadingIndicator;
+    private TextView resultTestTextView;
+    private TextView caseNameTextView;
+    private Button testButton;
+    private Boolean Testing = false;
+    private TestCase testCase;
+    private String testURL;
+
+
+    /**
+     * Constructs a new {@link TestCursorAdapter}.
+     *
+     * @param context The context
+     * @param c       The cursor from which to get the data.
+     */
+    public TestCursorAdapter(Context context, Cursor c) {
+        super(context, c, 0 /* flags */);
+    }
+
+    /**
+     * Makes a new blank list item view. No data is set (or bound) to the views yet.
+     *
+     * @param context app context
+     * @param cursor  The cursor from which to get the data. The cursor is already
+     *                moved to the correct position.
+     * @param parent  The parent to which the new view is attached to
+     * @return the newly created list item view.
+     */
+    @Override
+    public View newView(Context context, Cursor cursor, ViewGroup parent) {
+        // Inflate a list item view using the layout specified in list_item.xml
+        return LayoutInflater.from(context).inflate(R.layout.list_item, parent, false);
+
+
+    }
+
+    @Override
+    public void bindView(View view, Context context, Cursor cursor) {
+        caseNameTextView =(TextView)view.findViewById(R.id.case_name);
+        resultTestTextView=(TextView)view.findViewById(R.id.test_result);
+        testButton =(Button)view.findViewById(R.id.test_button);
+        loadingIndicator=(ProgressBar)view.findViewById(R.id.loading_indicator);
+
+        //Find the columns of testButton attributes that we're interested in
+        int nameColumnIndex = cursor.getColumnIndex(TestCaseEntry.COLUMN_TEST_NAME);
+        int controllerColumnIndex = cursor.getColumnIndex(TestCaseEntry.COLUMN_TEST_CONTROLLER);
+        int para1ColumnIndex=cursor.getColumnIndex(TestCaseEntry.COLUMN_TEST_PARA1);
+        int para2ColumnIndex=cursor.getColumnIndex(TestCaseEntry.COLUMN_TEST_PARA2);
+        int para3ColumnIndex=cursor.getColumnIndex(TestCaseEntry.COLUMN_TEST_PARA3);
+////
+        // Read the testButton attributes from the Cursor for the current testButton
+        String testName = cursor.getString(nameColumnIndex);
+        String testController = cursor.getString(controllerColumnIndex);
+        String testPara1=cursor.getString(para1ColumnIndex);
+        String testPara2=cursor.getString(para2ColumnIndex);
+//        String testPara3=cursor.getString(para3ColumnIndex);
+
+        testURL=testController+"/"+testPara1+"/"+testPara2;
+
+        Log.d(TAG, "bindView: this is from database \n "+"========"+testController+testPara1+testPara2);
+
+       caseNameTextView.setText(testURL);
+//        testCase=new TestCase("SensoryBoxAPK/AudioVolume/0.8", "200", "192.168.1.2", "8888");
+        testCase=new TestCase(testURL, "200", "192.168.1.2", "8888");
+        testButton.setOnClickListener(new View.OnClickListener() {
```

Comment:
Implement inside a custom class, so we can store and read the values for each view in its own instance.

Create the click listenet from inside the custom view so we have access to unique view data.